src/test/docker/fedora38/Dockerfile

```diff
@@ -0,0 +1,79 @@
+FROM fedora:38
+
+LABEL maintainer="Mark Grondona <mgrondona@llnl.gov>"
+
+#  Enable PowerTools for development packages
+RUN yum -y update \
+ && yum -y update \
+#  Utilities
+ && yum -y install \
+	wget \
+	man-db \
+	less \
+	git \
+	sudo \
+	munge \
+	ccache \
+	lua \
+	mpich \
+	valgrind \
+	jq \
+	which \
+	file \
+	vim \
+	patch \
+	diffutils \
+	hostname \
+#  Compilers, autotools
+	pkgconfig \
+	libtool \
+	autoconf \
+	automake \
+	gcc \
+	gcc-c++ \
+	libasan \
+	make \
+	cmake \
+#  Python
+	python36 \
```

Comment:
dumb question, if python 3.11 is the default python, why install 3.6?  I notice its there for some other fedora images.

Reply:
Good catch, I just copied the Fedora dockerfile (probably did that before too :facepalm:). Definitely should remove that before merging this!

Reply:
Fixed that, updated the `fluxrm/testenv:fedora38` image, and force-pushed here.